L80-csharp-load-balancing.md

```diff
@@ -0,0 +1,81 @@
+Title
+----
+* Author(s): James Newton-King
+* Approver: a11r
+* Status: Draft
+* Implemented in: csharp
+* Last updated: 2021-05-19
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+grpc-dotnet currently has no load balancing support. A channel is always created for a single address.
+
+This documentation outlines an initial load balancing implementation for the grpc-dotnet client.
+
+## Background
+
+grpc-dotnet currently doesn't support any form of load balancing in the client. The changes outlined in this proposal will enable basic gRPC load balancing and some of its related features. These include:
+
+* Name resolving. Built-in support is included for resolving from DNS, and a static resolver. Custom resolvers can be created.
+* Load balancing strategies. Built-in support is included for pick first and round robin load balancing. Custom load balancers can be created.
+* Connectivity status at sub-channel and channel level
+* Wait for ready
+
+## Proposal
+
+![Diagram](L80_graphics/diagram.png)
+
+### Initialization
+
+A channel is created with an address, an optional service provider, and an optional service config.
```

Comment:
Got it. Like other languages, the service config will be a fallback. If a resolver returns a service config then it will take precedence.

Reply:
Most languages actually have two channel args: one to set the default service config, and one to tell the resolver to never return a service config.  That way, the client API can be used to either set a default service config to be used only if the resolver does not return one, or it can be used to override the one that would have been returned by the resolver.

Reply:
I've added an argument to the channel options to disable the resolver getting a service config.